Branch-SDK/src/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java

```diff
@@ -149,16 +150,22 @@ private BranchResponse doRestfulPost(String url, JSONObject payload, int retryNu
                 retryNumber++;
                 return doRestfulPost(url, payload, retryNumber);
             } else {
+                InputStream inputStream = null;
                 try {
                     if (responseCode != HttpsURLConnection.HTTP_OK && connection.getErrorStream() != null) {
-                        return new BranchResponse(getResponseString(connection.getErrorStream()), responseCode);
+                        inputStream = connection.getErrorStream();
                     } else {
-                        return new BranchResponse(getResponseString(connection.getInputStream()), responseCode);
+                        inputStream = connection.getInputStream();
                     }
+                    return new BranchResponse(getResponseString(inputStream), responseCode);
                 } catch (FileNotFoundException ex) {
                     // In case of Resource conflict getInputStream will throw FileNotFoundException. Handle it here in order to send the right status code
                     PrefHelper.Debug("BranchSDK", "A resource conflict occurred with this request " + url);
                     return new BranchResponse(null, responseCode);
+                } finally {
+                    if (inputStream != null) {
```

Comment:
@aaaronlopez inputStream closing should be accompanied with try-catch for the crash prevention like this:

try { 
     if (inputStream != null) 
        inputStream.close(); 
     } catch(IOException e)